crates/sui-json-rpc/src/unit_tests/transaction_tests.rs

```diff
@@ -0,0 +1,318 @@
+        .await?;
+
+    assert_eq!(5, batch_responses.len());
+
+    for r in batch_responses.iter().skip(1) {
+        assert!(tx_responses
+            .iter()
+            .any(|resp| matches!(resp, SuiTransactionResponse {digest, ..} if *digest == r.digest)))
+    }
+
+    // test get_transactions_in_range_deprecated with smaller range
+    let tx: Vec<TransactionDigest> = http_client
+        .get_transactions_in_range_deprecated(1, 3)
+        .await?;
+    assert_eq!(2, tx.len());
+
+    // test get_transaction
+    for tx_digest in tx {
+        let response: SuiTransactionResponse = http_client
+            .get_transaction_with_options(
+                tx_digest,
+                Some(SuiTransactionResponseOptions::new().with_raw_input()),
+            )
+            .await?;
+        assert!(tx_responses.iter().any(
+            |resp| matches!(resp, SuiTransactionResponse {digest, ..} if *digest == response.digest)
+        ));
+        let sender_signed_data: SenderSignedData =
+            bcs::from_bytes(&response.raw_transaction).unwrap();
+        assert_eq!(sender_signed_data.digest(), tx_digest);
+    }
+
+    Ok(())
+}
+
+#[sim_test]
+async fn test_get_raw_transaction() -> Result<(), anyhow::Error> {
+    let cluster = TestClusterBuilder::new().build().await?;
+    let http_client = cluster.rpc_client();
+    let address = cluster.accounts.first().unwrap();
+
+    let objects = http_client
+        .get_owned_objects(
+            *address,
+            Some(SuiObjectResponseQuery::new_with_options(
+                SuiObjectDataOptions::new(),
+            )),
+            None,
+            None,
+            None,
+        )
+        .await?
+        .data;
+    let object_to_transfer = objects.first().unwrap().object().unwrap().object_id;
+
+    // Make a transfer transactions
+    let transaction_bytes: TransactionBytes = http_client
+        .transfer_object(*address, object_to_transfer, None, 1000, *address)
+        .await?;
+    let keystore_path = cluster.swarm.dir().join(SUI_KEYSTORE_FILENAME);
+    let keystore = Keystore::from(FileBasedKeystore::new(&keystore_path)?);
+    let tx = to_sender_signed_transaction(transaction_bytes.to_data()?, keystore.get_key(address)?);
+    let original_sender_signed_data = tx.data().clone();
+
+    let (tx_bytes, signatures) = tx.to_tx_bytes_and_signatures();
+
+    let response = http_client
+        .execute_transaction(
+            tx_bytes,
+            signatures,
+            Some(SuiTransactionResponseOptions::new().with_raw_input()),
+            Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+        )
+        .await?;
+
+    let decode_sender_signed_data: SenderSignedData =
+        bcs::from_bytes(&response.raw_transaction).unwrap();
+    // verify that the raw transaction data returned by the response is the same
+    // as the original transaction data
+    assert_eq!(decode_sender_signed_data, original_sender_signed_data);
+
+    Ok(())
+}
```

Comment:
@gegaowp per your previous comment, I added this test to assert decoded response is the same as original one